Xamarin.Forms.Platform.Android/Renderers/MotionEventHelper.cs

```diff
@@ -0,0 +1,53 @@
+using Android.Views;
+
+namespace Xamarin.Forms.Platform.Android
+{
+	internal class MotionEventHelper
+	{
+		VisualElement _element;
+		bool _isInViewCell;
+
+		public bool HandleMotionEvent(IViewParent parent)
+		{
+			if (_isInViewCell || _element.InputTransparent)
+			{
+				return false;
+			}
+
+			var renderer = parent as Platform.DefaultRenderer;
+			if (renderer == null)
+			{
+				return false;
+			}
+
+			// Let the container know that we're "fake" handling this event
+			renderer.NotifyFakeHandling();
+
+			return true;
+		}
+
+		public void UpdateElement(VisualElement element)
+		{
+			_isInViewCell = false;
+			_element = element;
```

Comment:
I worry we will be keeping this reference around since we don't clear or set to null our instance of MotionEventHelper.